### PR TITLE
Add a NULL check to EVP_PKEY_assign

### LINKED ISSUE #12619: EVP_PKEY_assign() bad input handling
I happen to have a test that verifies EVP_PKEY_assign_EC_KEY() & EVP_PKEY_assign_RSA() fail gracefully if passed a NULL pointer for the key.

It appears that sometime during 3.0.0 development the test broke (crashes) for the EC_KEY case only.

It looks to be a simple case of an unchecked de-reference.

This isn't an especially pressing issue given it's a negative testcase anyway, and I know there are many cases throughout the library where input pointers are not checked. This case jumped out both because it used to work & because different behavior is now observed depending on the key type.

### PR DESCRIPTION
**Pull request by jwalch**
Fixes  #12619

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

File: crypto/evp/p_lib.c [hunk 1]
```diff
@@ -716,7 +716,7 @@ int EVP_PKEY_assign(EVP_PKEY *pkey, int type, void *key)
     int alias = type;
 
 #ifndef OPENSSL_NO_EC
-    if (EVP_PKEY_type(type) == EVP_PKEY_EC) {
+    if (key && (EVP_PKEY_type(type) == EVP_PKEY_EC)) {
```
Comment (levitte) [inline crypto/evp/p_lib.c:719]: Please change `key` to `key != NULL`

**Review by levitte — CHANGES_REQUESTED** [review]

**Comment by mattcaswell** [comment]
Please can you submit a CLA? Or if you consider this to be a trivial contribution (from a copyright perspective) - please add the line "CLA: trivial" in a line on its own in the commit description.

**Comment by jwalch** [comment]
I sent in an ICLA to legal@opensslfoundation.org from the email address in question (jeremy.walch@gmail.com) more than a week ago now...

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot

**Comment by mattcaswell** [comment]
> I sent in an ICLA to legal@opensslfoundation.org from the email address in question (jeremy.walch@gmail.com) more than a week ago now...

Apologies. This somehow got over looked. I've processed it now.

**Review by mattcaswell — APPROVED** [review]
LGTM

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution.

**Pull request closed without merging**